mlir/lib/Dialect/EmitC/Transforms/FormExpressions.cpp

```diff
@@ -38,7 +38,7 @@ struct FormExpressionsPass
     auto matchFun = [&](Operation *op) {
       if (isa<emitc::CExpressionInterface>(*op) &&
           !op->getParentOfType<emitc::ExpressionOp>() &&
-          op->getNumResults() == 1)
+          !isa<emitc::LiteralOp>(op) && op->getNumResults() == 1)
```

Comment:
I was wondering about the ordering here, I think op->getNumResults() may be cheaper than getParentOfType (as it could keep walking until top) and even checking for interface is linear scan compared to checking if isa LiteralOp. Probably not crazy significant time wise though.